micro-rdk/src/common/ota.rs

```diff
@@ -259,28 +268,17 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         })
     }
 
-    pub(crate) async fn update(&mut self) -> Result<(), OtaError> {
-        let stored_metadata = if !self.storage.has_ota_metadata() {
-            log::info!("no ota metadata currently stored in NVS");
-            OtaMetadata::default()
-        } else {
-            self.storage
-                .get_ota_metadata()
-                .inspect_err(|e| log::warn!("failed to get ota metadata from nvs: {}", e))
-                .unwrap_or_default()
-        };
+    pub(crate) async fn needs_update(&self) -> bool {
+        self.stored_metadata().await.version != self.pending_version
+    }
 
-        if self.pending_version == stored_metadata.version() {
-            log::info!("firmware is up-to-date: `{}`", stored_metadata.version);
-            return Ok(());
+    /// Attempts to perform an OTA update.
+    /// On success, returns an `Ok(true)` or `Ok(false)` indicating if a reboot is necessary.
+    pub(crate) async fn update(&mut self) -> Result<bool, OtaError> {
```

Comment:
I don't particularly love the `bool` result. Maybe my unease is more to do with the name of the method?

Reply:
Would you prefer a field in `OtaService` that the caller checks after calling `update`?